src/kOS/Function/Suffixed.cs

```diff
@@ -308,4 +311,51 @@ public override void Execute(SharedObjects shared)
             shared.Cpu.PushStack(constants);
         }
     }
+    
+    [Function("allwaypoints")]
+    public class FunctionAllWaypoints : FunctionBase
+    {
+        public override void Execute(SharedObjects shared)
+        {
+            // ReSharper disable SuggestUseVarKeywordEvident
+            ListValue<WaypointValue> returnList = new ListValue<WaypointValue>();
+            // ReSharper ensable SuggestUseVarKeywordEvident
+
+            WaypointManager wpm = WaypointManager.Instance();
+            if (wpm == null)
+                shared.Cpu.PushStack(returnList); // When no waypoints exist, there isn't even a waypoint manager at all.
+
+            List<Waypoint> points = wpm.AllWaypoints();
```

Comment:
Ah yes you are correct.
change it to 
```
    if (wpm == null)
    {
        shared.Cpu.PushStack(returnList); // When no waypoints exist, there isn't even a waypoint manager at all.
        return;
    }
```
This was copied from code that was not originally a kOS function.  That line didn't push to the CPU stack, it just did a return directly.  I changed it to a push to the stack but forgot that this would mean it falls through and continues on.